chart/templates/postgres.yaml

```diff
@@ -20,11 +20,24 @@ spec:
           emptyDir:
             medium: Memory
             sizeLimit: "{{- .Values.db.shmVolume  }}"
-        - name: conf
-          configMap:
-            name: postgresql-conf
-            optional: true
       initContainers:
+        {{- if .Values.db.upgrade.perform}}
+        - image: tianon/postgres-upgrade:{{.Values.db.upgrade.old}}-to-{{.Values.db.upgrade.new}}
+          name: postgres-upgrade
+          volumeMounts:
+            - mountPath: "/current"
+              name: postgresql
+          command:
+            - sh
+            - -c
+            - "if [ ! -d '/current/postgres' ]; then exit 0; fi && rm -rf /var/lib/postgresql/{{.Values.db.upgrade.old}}/data &&
```

Comment:
Can we make this a no op if the db version is already on new ?

Reply:
The upgrade script does that and exits

Reply:
added